test/libsolidity/ParserTest.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Unit tests for the parser
+ */
+
+#include <test/Options.h>
+
+#include <liblangutil/SourceLocation.h>
+#include <libevmasm/Assembly.h>
+
+#include <liblangutil/Scanner.h>
+#include <libsolidity/parsing/Parser.h>
+#include <libsolidity/analysis/NameAndTypeResolver.h>
+#include <libsolidity/codegen/Compiler.h>
+#include <libsolidity/ast/AST.h>
+#include <libsolidity/analysis/TypeChecker.h>
+#include <liblangutil/ErrorReporter.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <string>
+#include <iostream>
+
+using namespace std;
+using namespace langutil;
+using namespace dev::eth;
+
+namespace dev
+{
+namespace solidity
+{
+class Contract;
+namespace test
+{
+
+namespace
+{
+
+ASTPointer<SourceUnit> parseContract(std::shared_ptr<CharStream> _sourceCode)
+{
+	ErrorList errors;
+	ErrorReporter errorReporter(errors);
+	Parser parser(errorReporter);
+	ASTPointer<SourceUnit> sourceUnit;
+	BOOST_REQUIRE_NO_THROW(sourceUnit = parser.parse(make_shared<Scanner>(_sourceCode)));
+	BOOST_CHECK(!!sourceUnit);
+
+	return sourceUnit;
+}
+
+} // end anonymous namespace
+
+BOOST_AUTO_TEST_SUITE(Parser)
```

Comment:
Why didn't you add this to SolidityParser.cpp?

Reply:
I must have misunderstood your suggestion :)